Tests/DatadogIntegrationTests/TracingIntegrationTests.swift

```diff
@@ -32,11 +32,9 @@ class TracingIntegrationTests: IntegrationTests {
         )
         app.tapSendTracesForUITests()
 
-        // Wait for delivery
-        Thread.sleep(forTimeInterval: Constants.dataDeliveryTime)
+        // Return desired count or timeout
+        let recordedTracingRequests = try tracingServerSession.pullRecordedPOSTRequests(count: 1, timeout: Constants.dataDeliveryTime)
 
-        // Assert tracing requests
-        let recordedTracingRequests = try tracingServerSession.getRecordedPOSTRequests()
```

Comment:
There are also two other places in this `TracingIntegrationTests` where we use `.getRecordedPOSTRequests()` - let's change them to `.pullRecordedPOSTRequests()` 💪.

Explanation: we use 3 different server sessions in this test:
* one for recording traces,
* one for recording logs (send by `span.log()`),
* one for recording data request which should be auto-instrumented with trace propagation HTTP headers.

Reply:
Yes, I saw them, but they were not waiting a fixed time before calling `.getRecordedPOSTRequests()` thats why I didn't change them, should I?

Reply:
Yeah, it works, because data send to that sessions is delivered before data send to `tracingServerSession`. But this is an implicit implementation-based assumption, so I think by using `.pullRecordedPOSTRequests()` we will make it more explicit. Anyway, it will still keep working, so both ways are good for me 👍.

Reply:
Ok, the I will change them also